urllib3/poolmanager.py

```diff
@@ -129,22 +129,25 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # Return False to re-raise any potential exceptions
         return False
 
-    def _new_pool(self, scheme, host, port):
+    def _new_pool(self, scheme, host, port, pool_kwargs=None):
         """
-        Create a new :class:`ConnectionPool` based on host, port and scheme.
+        Create a new :class:`ConnectionPool` based on host, port, scheme, and
+        any additional pool keyword arguments.
 
-        This method is used to actually create the connection pools handed out
-        by :meth:`connection_from_url` and companion methods. It is intended
-        to be overridden for customization.
+        If `pool_kwargs` is provided, it overrides the instance's
+        `connection_pool_kw` variable. This method is used to actually create
```

Comment:
I was on the fence about merging vs overriding and went with overriding because it seemed like a simple way to avoid the unsetting problem. Is it more of a burden on the user to make sure they have all the values they need in the override dictionary, or to make sure they correctly unset values we're merging in? I honestly don't know, so if merging is more appealing to you, I can change it to merge.